### ISSUE TITLE
Predefined padding loss in accuracy

### ISSUE THREAD

**Issue by armandmcqueen** [armandmcqueen/tensorpack-mask-rcnn#50]
When we use PREDEFINED_PADDING, we lose accuracy. This is a pure throughput optimization and the loss in accuracy needs to be address.

PREDEFINED_PADDING is when we have a small number of acceptable image shapes and we pad all images to fit into one of those shapes. This reduces the number of input shapes, speeding up cudnn autotuning from 8+ epochs to ~3 epochs, leading to a good improvement in time-to-train. 

PREDEFINED_PADDING is focused on COCO - it is not useful if your input images are already all the same size.

**Comment by armandmcqueen** [comment]
Tested and confirmed that predefined_padding does not hurt accuracy. Looked like we used different learning rates between tests, which made it look like predefined_padding hurts accuracy.
